capsules/src/driver.rs

```diff
@@ -37,3 +41,44 @@ pub enum NUM {
     USB_USER = 0x20005,
 }
 }
+
+// sorted in order of ascending driver number
+// this is so it's easier to avoid conflicts when adding a new driver
+pub fn get_permission_bit(driver_num: usize) -> Option<usize> {
+    if let Some(num) = NUM::from_usize(driver_num) {
+        Some(match num {
+            NUM::ALARM => 0,
+            NUM::CONSOLE => 1,
+            NUM::LED => 2,
+            NUM::BUTTON => 3,
+            NUM::GPIO => 4,
+            NUM::ADC => 5,
+            NUM::DAC => 6,
+            NUM::ANALOG_COMPARATOR => 7,
+            NUM::SPI => 8,
+            NUM::USB_USER => 9,
+            NUM::I2C_MASTER_SLAVE => 10,
+            NUM::BLE_ADVERTISING => 11,
+            NUM::RNG => 12,
+            NUM::CRC => 13,
+            NUM::I2C_MASTER => 14,
+            NUM::APP_FLASH => 15,
+            NUM::NVM_STORAGE => 16,
+            NUM::SD_CARD => 17,
+            NUM::TEMPERATURE => 18,
```

Comment:
@alevy With regards to a mapping of bits to driver numbers being huge because, for example, the driver number for temperature is `0x60000`, this is where the additional mapping takes place. As I have it, `0x60000` maps to `1 << 18`. That being said, it doesn't mean that Phil's suggestion might not be a better way -- just that I had already created the mapping that prevents the need for half a million bits.